### ISSUE TITLE
cmake_multi does not use Release lib for RelWithDebInfo

### ISSUE THREAD

**Issue by boussaffawalid** [conan-io/conan#5131]
Hello,

I'm using cmake_multi generator to build my project using visual studio.
I noticed that cmake_multi does not handel ReleaseWithDebInfo nor MinSizeRel

An example from my conanbuildinfo_multi.cmake

```
add_library(CONAN_PKG::my_package INTERFACE IMPORTED)

# Property INTERFACE_LINK_FLAGS do not work, necessary to add to INTERFACE_LINK_LIBRARIES
set_property(TARGET CONAN_PKG::my_package PROPERTY INTERFACE_LINK_LIBRARIES ${CONAN_PACKAGE_TARGETS_MY_PACKAGE} ${CONAN_SHARED_LINKER_FLAGS_MY_PACKAGE_LIST} ${CONAN_EXE_LINKER_FLAGS_MY_PACKAGE_LIST}
                                                             $<$<CONFIG:Release>:${CONAN_PACKAGE_TARGETS_MY_PACKAGE_RELEASE} ${CONAN_SHARED_LINKER_FLAGS_MY_PACKAGE_RELEASE_LIST} ${CONAN_EXE_LINKER_FLAGS_MY_PACKAGE_RELEASE_LIST}>
                                                             $<$<CONFIG:RelWithDebInfo>:${CONAN_PACKAGE_TARGETS_MY_PACKAGE_RELEASE} ${CONAN_SHARED_LINKER_FLAGS_MY_PACKAGE_RELEASE_LIST} ${CONAN_EXE_LINKER_FLAGS_MY_PACKAGE_RELEASE_LIST}>
                                                             $<$<CONFIG:MinSizeRel>:${CONAN_PACKAGE_TARGETS_MY_PACKAGE_RELEASE} ${CONAN_SHARED_LINKER_FLAGS_MY_PACKAGE_RELEASE_LIST} ${CONAN_EXE_LINKER_FLAGS_MY_PACKAGE_RELEASE_LIST}>
                                                             $<$<CONFIG:Debug>:${CONAN_PACKAGE_TARGETS_MY_PACKAGE_DEBUG} ${CONAN_SHARED_LINKER_FLAGS_MY_PACKAGE_DEBUG_LIST} ${CONAN_EXE_LINKER_FLAGS_MY_PACKAGE_DEBUG_LIST}>
                                                             )
```

Basically, cmake multi generator sets RelWithDebInfo and CONFIG:MinSizeRel libs to Release.

The same issue when loading config files:

```
if(EXISTS ${CMAKE_CURRENT_LIST_DIR}/conanbuildinfo_release.cmake)
    include(${CMAKE_CURRENT_LIST_DIR}/conanbuildinfo_release.cmake)
else()
    message(FATAL_ERROR "No conanbuildinfo_release.cmake, please install the Release conf first")
endif()

if(EXISTS ${CMAKE_CURRENT_LIST_DIR}/conanbuildinfo_debug.cmake)
    include(${CMAKE_CURRENT_LIST_DIR}/conanbuildinfo_debug.cmake)
else()
    message(FATAL_ERROR "No conanbuildinfo_debug.cmake, please install the Debug conf first")
endif()
```

conanbuildinfo_RelWithDebInfo.cmake and conanbuildinfo_MinSizeRel.cmake are not included.

Is it intended to be this way ?  
How can I link RelWithDebInfo library ?

**Comment by lasote** [comment]
Yes, I think you are right. @memsharded Any feedback about it? We made the `cmake_find_package_multi` already prepared for `RelWithDebInfo` and `MinSizeRel`. Could it be considered breaking? I think this is a bug, but please @memsharded tell us.

**Comment by lukastvrdy** [comment]
I'm affected by this problem too with cmake_multi.

**Comment by memsharded** [comment]
@boussaffawalid @lukastvrdy fixed by #5221, will be released next Conan 1.16.
